utilities/menu-jump.js

```diff
@@ -0,0 +1,59 @@
+/* Copyright (c) 2015-present, salesforce.com, inc. All rights reserved */
+/* Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license */
+// import ReactDOM from 'react-dom';
+import escapeRegExp from 'lodash.escaperegexp';
+
+const menuJump = ({
+	container,
+	itemTag = 'li',
+	key,
+	keyBuffer,
+	keyCode,
+	options,
+	scrollToFocusedIndex = true,
+}) => {
+	let ch = key || String.fromCharCode(keyCode);
+
+	if (/^[ -~]$/.test(ch)) {
+		ch = ch.toLowerCase();
+	} else {
+		ch = null;
+	}
+
+	const pattern = keyBuffer(ch);
+	let consecutive = 0;
+	let focusedIndex;
+
+	// Support for navigating to the next option of the same letter with repeated presses of the same key
+	if (
+		pattern.length > 1 &&
+		new RegExp(`^[${escapeRegExp(ch)}]+$`).test(pattern)
+	) {
+		consecutive = pattern.length;
+	}
+
+	options.forEach((item, index) => {
+		const itemLabel = item.label.toLowerCase();
```

Comment:
`item.label` is supposed to be limited to a string right now, but we should protect this, and make sure it's a string before calling `toLowerCase`.